pkg/namespacecheck/namespace_info.go

```diff
@@ -0,0 +1,268 @@
+package namespacecheck
+
+import (
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// TODO these could be hardcoded in a different package or repo and simply checked here.
+var (
+	AllKnownNamespaces = sets.NewString(
+		"default",
+		"kube-system",
+		"openshift",
+		"openshift-apiserver",
+		"openshift-apiserver-operator",
+		"openshift-authentication",
+		"openshift-authentication-operator",
+		"openshift-cloud-controller-manager",
+		"openshift-cloud-controller-manager-operator",
+		"openshift-cloud-credential-operator",
+		"openshift-cloud-network-config-controller",
+		"openshift-cluster-csi-drivers",
+		"openshift-cluster-machine-approver",
+		"openshift-cluster-node-tuning-operator",
+		"openshift-cluster-samples-operator",
+		"openshift-cluster-storage-operator",
+		"openshift-cluster-version",
+		"openshift-config",
+		"openshift-config-managed",
+		"openshift-config-operator",
+		"openshift-console",
+		"openshift-console-operator",
+		"openshift-controller-manager",
+		"openshift-controller-manager-operator",
+		"openshift-dns",
+		"openshift-dns-operator",
+		"openshift-etcd",
+		"openshift-etcd-operator",
+		"openshift-host-network",
+		"openshift-image-registry",
+		"openshift-ingress",
+		"openshift-ingress-canary",
+		"openshift-ingress-operator",
+		"openshift-insights",
+		"openshift-kni-infra",
+		"openshift-kube-apiserver",
+		"openshift-kube-apiserver-operator",
+		"openshift-kube-controller-manager",
+		"openshift-kube-controller-manager-operator",
+		"openshift-kube-scheduler",
+		"openshift-kube-scheduler-operator",
+		"openshift-kube-storage-version-migrator",
+		"openshift-kube-storage-version-migrator-operator",
+		"openshift-machine-api",
+		"openshift-machine-config-operator",
+		"openshift-marketplace",
+		"openshift-monitoring",
+		"openshift-multus",
+		"openshift-network-diagnostics",
+		"openshift-network-operator",
+		"openshift-nutanix-infra",
+		"openshift-oauth-apiserver",
+		"openshift-openstack-infra",
+		"openshift-operator-lifecycle-manager",
+		"openshift-operators",
+		"openshift-ovirt-infra",
+		"openshift-ovn-kubernetes",
+		"openshift-service-ca",
+		"openshift-service-ca-operator",
+		"openshift-user-workload-monitoring",
+		"openshift-vsphere-infra",
+	)
+
+	NamespacesToJiraComponents = map[string]string{
```

Comment:
Is the requirement then that people add the namespace to their component, here and also down below? I'm not sure I like that - what does it buy us for the added complexity?

I feel like components should be the canonical owners of their configuration.  Couldn't you generate these maps by asking the components for their namespaces?

Reply:
I could see wanting this if someone else wants to import this to known ownership of a namespace.  Could we generate this file with `go generate` instead?